strategy/from_json/to_object_containing.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
+
+import { JSONArray, JSONValue, Serializable } from "../../serializable.ts";
+import { FromJSONStrategy } from "../compose_strategy.ts";
+import {
+  ERROR_TO_OBJECT_CONTAINING_INVALID_SUB_VALUE,
+  ERROR_TO_OBJECT_CONTAINING_INVALID_VALUE,
+  ERROR_TO_OBJECT_CONTAINING_USE_TO_SERIALIZABLE,
+} from "../../error_messages.ts";
+
+/** revive data from `{k: v}` using `fromJSON` on a subclass type `v` */
+export function toObjectContaining<T>(
+  type: T & { new (): Serializable },
+): FromJSONStrategy {
+  return function _toObjectContaining(
+    value: JSONValue,
+  ) {
+    if (value == null) {
+      return null;
+    }
+
+    if (Array.isArray(value)) {
+      throw Error(ERROR_TO_OBJECT_CONTAINING_USE_TO_SERIALIZABLE);
+    }
+
+    if (Object.prototype.toString.call(value) !== "[object Object]") {
```

Comment:
Is this the best way to check is a given value is a POJO object? Maybe a comment here could help.

Reply:
Yea it is, I know it's expensive